elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -90,6 +90,11 @@ default void preCommit() {
      */
     <T> T loadObject(Class<T> entityClass, Serializable id);
 
+    default <T> T loadObject(Class<T> entityClass, Serializable id, FilterScope filterScope) {
```

Comment:
Instead of having FilterScope here, why not make this more forward looking and directly pass the FilterExpression.